utils/test_results.py

```diff
@@ -0,0 +1,74 @@
+import datetime as dt
+
+from django.contrib.postgres.aggregates import ArrayAgg
+from django.db.models import Avg, Case, FloatField, Value, When
+from shared.django_apps.reports.models import Test, TestInstance
+
+thirty_days_ago = dt.datetime.now(dt.UTC) - dt.timedelta(days=30)
+
+
+def aggregate_test_results(
```

Comment:
How does this function work with pagination?

Reply:
currently it does not, but i see that we'd want to paginate by test, like get the first 10 tests then the next 10 tests